docs/bugs.json

```diff
@@ -1,4 +1,12 @@
 [
+	{
+		"name": "DynamicArrayCleanup",
+		"summary": "When assigning a dynamic-array in storage, whose types are packed into a slot, certain part of slots were not zeroed out.",
+		"description": "Consider a dynamic-array in storage whose base-type can be packed into a single slot, such as `uint128[]`. Let us define its length to be `l`. When this array gets assigned by another dynamic array with a smaller length, say `m`, the slots between `l` and `m` has to be cleaned by zeroing them out. However, this cleaning was not performed properly. Specifically, if `k` is the number of elements that are packed into a single slot, then from slots `m/k + 1` until `l/k`,  only the first packed value was cleaned up. If this array gets resized to a length larger than `m`, the indices corresponding to the unclean parts of the slot contained the old value, instead of 0. The resizing here is performed by assigning to the array `slot` or assigning to the array `length` or by a `push()`.",
+		"introduced": "0.4.11",
+		"fixed": "0.7.3",
+		"severity": "medium"
```

Comment:
I'm wondering whether this might even qualify as "high" actually... I guess the justification for "medium" is that it's not extremely likely to occur? Because *if* it occurs, it is severe...

Reply:
It might even be that people did not really assume slots to be zeroed out. So It could be that this bug has been discovered at some point, but rather thought to be a feature than a bug...

Reply:
I thought about it some more - I think the most likely scenario for pushing something empty is "reserving space" for something that is later referred to by its index into an array - but that will generally be something larger than 32 bytes and not affected by this. Apart from that, given that apparently nobody ever noticed since 0.4.11, it may be rather unusual indeed, so "medium" may be fine.